src/components/widgets/camera/services/WebrtcGo2RtcCamera.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+  <video
+    ref="streamingElement"
+    autoplay
+    playsinline
+    controls
+    muted
+    :style="cameraStyle"
+    :crossorigin="crossorigin"
+  />
+</template>
+
+<script lang="ts">
+import { Component, Ref, Mixins } from 'vue-property-decorator'
+import CameraMixin from '@/mixins/camera'
+
+@Component({})
+export default class WebrtcGo2RtcCamera extends Mixins(CameraMixin) {
+  @Ref('streamingElement')
+  readonly cameraVideo!: HTMLVideoElement
+
+  pc: RTCPeerConnection | null = null
+  remoteId: string | null = null
+  ws: WebSocket | null = null
+
+  // webrtc player methods
+  // adapted from https://github.com/AlexxIT/go2rtc/blob/master/www/webrtc.html
+  // also adapted from https://github.com/mainsail-crew/mainsail/pull/1651
+
+  get url () {
+    const urlSearch = new URL(this.buildAbsoluteUrl(this.camera.urlStream || '')).search.toString()
+    const url = new URL('api/ws' + urlSearch, this.buildAbsoluteUrl(this.camera.urlStream || ''))
+    url.searchParams.set('media', 'video+audio')
+    // change protocol to ws
+    url.protocol = url.protocol === 'https'
+      ? 'wss:'
+      : 'ws:'
+    return url.toString()
+  }
```

Comment:
I am struggling to see what the point of this bit is here... the whole objective is that users supply the stream URL, so in this case I think should be "ws://<host>:<port>/api/ws?src=<camera>&media=video+audio" so that this method would not be required.

Reply:
go2rtc doesn't obviously expose the wss/ws urls, it exposes only the http url links, so from a user perspective its easier to get those.

Reply:
example this is the streams page for a given stream.

![image](https://github.com/fluidd-core/fluidd/assets/152501/61c0288b-54b1-4641-bdcd-324620b262ac)

Reply:
I've commented on the original Mainsail PR on this topic so we can get to a common ground here.

Reply:
updated your if statement missing colon! broke https support!

Reply:
That's what I get for coding in GitHub comments! 😅

Reply:
and me for reviewing it in!